scripts/setup.sh

```diff
@@ -0,0 +1,20 @@
+# Check if Homebrew exists, install it if not.
+which -s brew
+if [[ $? != 0 ]] ; then
+    echo "Homebrew not installed, installing..."
+    ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"
+else
+    echo "Homebrew already installed, checking for update..."
+    brew update
+fi
+
+# Check if Go exists, install it if not.
```

Comment:
I don't think you're actually checking to see if go exists? `$?` doesn't read anything.

Does `go version` suffice? This should error out if go isn't installed.